src/components/Modal/auto/ModalAuto.tsx

```diff
@@ -1,35 +1,16 @@
 import React, { memo, useEffect, useState } from 'react'
 
-import { Button, ButtonProps } from '../../Button'
-import { HFlow } from '../../HFlow'
-import { Modal, ModalDepthLevel, ModalProps } from '../Modal'
+import { ButtonProps } from '../../Button'
+import { Modal, ModalProps } from '../Modal'
 import { ModalBody } from '../ModalBody'
 import { ModalFooter } from '../ModalFooter'
+import { HFlow } from '../../HFlow'
+import { ModalAutoFooterButton } from './ModalAutoFooterButton'
 
 export type ButtonAction = ButtonProps & { label?: React.ReactNode; ['data-testid']?: string }
 
-export interface ModalAutoProps {
+export interface ModalAutoProps extends Omit<ModalProps, 'open'> {
   actions?: ButtonAction[]
-  size?: ModalProps['size']
-
-  /**
-   * @description allows you to customize the depth of the container and the backdrop of the modal
-   * @default 1 - the lowest possible value
-   */
-  depthLevel?: ModalDepthLevel
```

Comment:
Why did you remove this one?

Reply:
Because these props are the same as the ones from `Modal`. They were duplicated here, and I didn’t see a reason not to inherit them along with the rest of the `Modal` props. So I manually removed the duplicated ones and kept everything under the `extends` in the interface.